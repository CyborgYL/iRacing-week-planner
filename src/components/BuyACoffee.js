// @flow

import axios from 'axios';
import * as React from 'react';

import { useTranslation } from 'react-i18next';
import styles from './styles/buyACoffee.module.css';

export default function BuyACoffee(): React.Node {
  const { t, i18n } = useTranslation();
  const getBCUrl = async () => {
    let response;
    try {
      response = await axios.get('https://geolocation-db.com/json/09068b10-55fe-11eb-8939-299a0c3ab5e5', {
        headers: { Accept: '' }, // Avoid OPTION
      });
    } catch (error) {
      if (i18n.language === 'en-GB') {
        return 'https://www.buymeacoffee.com/iwpgbp';
      }
      if (['es', 'fr', 'de'].includes(i18n.language)) {
        return 'https://www.buymeacoffee.com/iwpeur';
      }
      return 'https://www.buymeacoffee.com/tmoitie';
    }

    if (response.data.country_code === 'GB') {
      return 'https://www.buymeacoffee.com/iwpgbp';
    }

    const euroCountries = [
      'AT', 'BE', 'CY', 'EE', 'FI', 'FR', 'DE', 'GR', 'IE', 'IT', 'LV', 'LT', 'LU', 'MT', 'NL', 'PT', 'ES', 'SI', 'SK',
      'XK', 'ME', 'AD', 'MC', 'SM', 'VA',
    ];

    if (euroCountries.includes(response.data.country_code)) {
      return 'https://www.buymeacoffee.com/iwpeur';
    }

    return 'https://www.buymeacoffee.com/tmoitie';
  };

  const clickBuyCoffee = async () => {
    window.location.assign(await getBCUrl());
  };

  return (
    <button type="button" className={styles.bmcButton} onClick={clickBuyCoffee} onKeyPress={clickBuyCoffee}>
      <img
        src="https://cdn.buymeacoffee.com/buttons/bmc-new-btn-logo.svg"
        alt={t('Buy me a coffee')}
      />
      <span style={{ marginLeft: '5px' }}>
        {t('Buy me a coffee')}
      </span>
    </button>
  );
}
