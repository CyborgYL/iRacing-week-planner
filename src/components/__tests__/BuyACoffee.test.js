import React from 'react';
import { shallow } from 'enzyme';
import { describe, test } from '@jest/globals';

import BuyACoffee from '../BuyACoffee';

describe('components/BuyACoffee', () => {
  test('renders correctly', () => {
    const component = shallow(<BuyACoffee />);

    expect(component).toMatchSnapshot();
  });
});
