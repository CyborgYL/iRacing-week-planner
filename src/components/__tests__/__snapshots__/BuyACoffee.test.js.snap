// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/BuyACoffee renders correctly 1`] = `
<a
  className="bmcButton"
  href="https://www.buymeacoffee.com/tmoitie"
  rel="noreferrer"
  target="_blank"
>
  <img
    alt="Buy me a coffee"
    src="https://cdn.buymeacoffee.com/buttons/bmc-new-btn-logo.svg"
  />
  <span
    style={
      Object {
        "marginLeft": "5px",
      }
    }
  >
    Buy me a coffee
  </span>
</a>
`;
